app/models/filter.rb

```diff
@@ -135,6 +141,21 @@ def expire_topbar_cache(sweeper)
     role.usergroups.each { |g| g.expire_topbar_cache(sweeper) }
   end
 
+  def disable_overriding!
+    self.override = false
+    self.save!
+  end
+
+  def inherit_taxonomies
+    inherit_taxonomies! unless self.override?
```

Comment:
How about calling `before_save :inherit_taxonomies!, :unless => -> { |filter| filter.override }` so you can save writing this method? It doesn't do anything other than the `unless` check that you can perform in the `before_save` call